### PR TITLE
Features/functionality and testing

### PR DESCRIPTION
**Pull request by CalebCyphers**
### What does this PR do?

- Breaks the random joke display out into it's own component: RandomJoke
- Adds Conditionally rendered 'Return to home' button
- Adds unit tests using react-testing-library to RandomJoke.test
- Adds Cypress to project
- Adds first passing test to Cypress

### Where should the reviewer start?

- Cypress/integration/App_spec.js

### Any background context you want to provide?

- be sure to run npm install to get Cypress up and running when you pull down

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -29,37 +31,35 @@ class App extends Component {
     }
   }
 
+  homeButtonDisplayUpdater = (bool) => {
```
Comment (peytonjo) [inline src/App/App.js:34]: may need to rename this function name its a bit long

**Review by peytonjo — APPROVED** [review]
looks great!

File: src/App/App.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+
 class App extends Component {
   constructor() {
     super();
     this.state = {
       joke: '',
       id: '',
-      error: ''
+      error: '',
+      displayHomeButton: false
```
Comment (jdxsmith) [inline src/App/App.js:18]: Nice use of state to determine if something is displayed

File: src/App/App.test.js [hunk 3]
```diff
@@ -1,8 +1,71 @@
...
+  it("renders the Navbar and RandomJoke", async () => {
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+
+    const title = screen.getByText("Cringy Dad Jokes");
+    const addNewJokeBtn = screen.getByText("Add new joke")
+    // const firstJoke = await waitFor(() => screen.getByText("Yo mamma"));
```
Comment (jdxsmith) [inline src/App/App.test.js:34]: Let's look further into testing the render of our joke

**Review by jdxsmith — APPROVED** [review]
Nice work, let's just make sure to continue to build up our testing coverage.

**Pull request merged**